adapt/tools/text/trie.py

```diff
@@ -30,15 +30,19 @@ def __init__(self, data=None, is_terminal=False):
 
     def lookup(self, iterable, index=0, gather=False, edit_distance=0, max_edit_distance=0, match_threshold=0.0, matched_length=0):
         """
-        TODO: Implement trie lookup with edit distance
-
         Args:
-            iterable(list?): key used to find what is requested this could
-                be a generator.
-            index(int): index of what is requested
-            gather(bool): of whether to gather or not
-            edit_distance(int): the distance -- currently not used
-            max_edit_distance(int): the max distance -- not currently used
+            iterable(hashable): a list of items used to traverse the Trie
```

Comment:
Maybe a single line comment as well on the lines of

"A generator function traversing the Trie returning matching objects".

so the purpose is explicit and not inferred by name and parameters.


Also Is the iterable the "lookup criteria"? "a list of items used to traverse the Trie" doesn't really tell me that this is where I put in a sequence identifying a node in the trie. I understand that this is quite generic but maybe we could add examples...  "such as a string of letters or a sequence of numbers"